crates/relayer/src/main.rs

```diff
@@ -174,39 +186,105 @@ impl RelayerService {
         // Create blockchain client from relayer config
         let transports = self.create_client()?;
 
-        // Clone mock relayer for the async task
+        // Clone mock relayer and metrics for the async task
         let mock_relayer = self.mock_relayer.clone();
+        let metrics = self.metrics.clone();
 
         let handle = async move {
             while let Some((request, res_tx)) = rx.recv().await {
+                // Record queue depth (items waiting to be processed)
+                // rx.recv() has already removed the current item, so rx.len() is correct
+                if let Some(ref metrics) = metrics {
+                    metrics.set_queue_depth(rx.len() as i64);
+                }
+
+                let protocol_name = request.protocol.to_string();
+                if let Some(ref metrics) = metrics {
+                    metrics.inc_protocol_requests(&protocol_name);
+                }
```

Comment:
Same, possible cardinality explosion as the metric is recorded before the code checks if the protocol is supported/enabled. You can crash the relayer by flooding it with requests specifying random protocol names.